graphql/resolvers.go

```diff
@@ -3284,10 +3283,26 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 			}
 		}
 		mainlineCommitVersion := MainlineCommitVersion{}
-
+		shouldCollapse := false
+		if !utility.FromBoolPtr(v.Activated) {
+			shouldCollapse = true
+		} else if buildVariantOptions != nil && (len(buildVariantOptions.Tasks) > 0 || len(buildVariantOptions.Variants) > 0 || len(buildVariantOptions.Statuses) > 0) {
```

Comment:
Can we have HasMatchingTasks handle this (since it's implicitly true that it does if we aren't filtering)? Or at least make a buildVariantOptions helper method for this, like IsPopulated()